Automattic/jetpack
# Autoloader: Load only latest version of autoload files to avoid conflicts.

Fixes #14389

#### Changes proposed in this Pull Request:

* Register each autoload file declaration and keep track of versions in the same way we do for classes. Only after all plugins have registered their packages and plugins loaded is fired should we attempt to include the files so that only the latest version is loaded.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
No, just a bug fix should cause no BC breaks.

#### Testing instructions:
* See issue for plugin to test with.
* Activate both and obverse no errors
* Also try other plugins which use jetpack autoloader to ensure they still fuction.
* Trying as many different composer packages is helpful

#### Proposed changelog entry for your changes:
* Fix: Conflicts which can occur when jetpack autoloader includes autoload files in the wrong order.
